### PR TITLE
Fix external environment simulation

### PR DESCRIPTION
**Pull request by mjpatell**

### CODE REVIEW

File: cob_gazebo_worlds/launch/world.launch [hunk 1]
```diff
@@ -16,10 +16,13 @@
 	<env name="GAZEBO_MODEL_PATH" value="$(arg pkg_gazebo_worlds)/Media/models:$(optenv GAZEBO_MODEL_PATH)"/>
 
 	<!-- upload default configuration parameters -->
-	<include file="$(arg pkg_env_config)/upload_navigation_goals.launch">
+	<include file="$(find cob_default_env_config)/upload_navigation_goals.launch">
+		<arg name="pkg_env_config" value="$(arg pkg_env_config)"/>
```
Comment (fmessmer) [inline cob_gazebo_worlds/launch/world.launch:20]: please revert these lines and add a `upload_navigation_goals.launch` to your external `pkg_env_config` package

File: cob_gazebo_worlds/launch/world.launch [hunk 2]
```diff
@@ -16,10 +16,13 @@
...
 		<arg name="robot_env" value="$(arg robot_env)" />
 	</include>
-	<include file="$(arg pkg_env_config)/upload_object_locations.launch">
+	
+	<include file="$(find cob_default_env_config)/upload_object_locations.launch">
+		<arg name="pkg_env_config" value="$(arg pkg_env_config)"/>
```
Comment (fmessmer) [inline cob_gazebo_worlds/launch/world.launch:25]: please revert these lines and add a `upload_object_locations.launch` to your external `pkg_env_config` package

File: cob_gazebo_worlds/launch/world.launch [hunk 3]
```diff
@@ -34,6 +37,8 @@
 	</include>
 
 	<!-- spawn environment and start environment-specific nodes -->
-	<include file="$(arg pkg_gazebo_worlds)/launch/$(arg robot_env).xml"/>
+	<include file="$(arg pkg_gazebo_worlds)/launch/$(arg robot_env).xml">
+		<arg name="pkg_gazebo_worlds" value="$(arg pkg_gazebo_worlds)"/>
+	</include>
```
Comment (fmessmer) [inline cob_gazebo_worlds/launch/world.launch:42]: please revert these lines and add a `launch/$(arg robot_env).xml` to your external `pkg_env_config` package

then you do not need to pass `pkg_gazebo_worlds` to all the other `*.xml` files in this package

**Review by fmessmer — COMMENTED** [review]
please consider my comments...

Should be squash-merged thereafter!

**Review by fmessmer — CHANGES_REQUESTED** [review]
see above

File: cob_gazebo_worlds/launch/world.launch (hunk 1 above)
Comment (mjpatell) [reply cob_gazebo_worlds/launch/world.launch:20]: In order to avoid copy of that file in the external package. I added this way. Is this wrong?

Comment (fmessmer) [reply cob_gazebo_worlds/launch/world.launch:20]: well, it adds an argument to a lot of files without being used there...

maybe we should call the `upload_navigation_goal.launch` and `upload_object_locations.launch` directly from `cob_bringup_sim/launch/robot.launch`...
this way, we could actually get rid of the dependency to `cob_default_env_config` within `cob_gazebo_worlds`

Comment (fmessmer) [reply cob_gazebo_worlds/launch/world.launch:20]: the unused argument also breaks the launch file checks...I'll propose something...

Comment (mjpatell) [reply cob_gazebo_worlds/launch/world.launch:20]: Our stuff works fine from our environment. So you can close this request.
Thank you.

**Comment by fmessmer** [comment]
happy to hear...

**Pull request closed without merging**